app/common/data/interfaces/collections.py

```diff
@@ -429,6 +497,21 @@ def add_question_condition(question: Question, user: User, managed_expression: "
 
     expression = Expression.from_managed(managed_expression, user)
     question.expressions.append(expression)
+
+    if (
+        isinstance(managed_expression, BaseDataSourceManagedExpression)
+        and managed_expression.referenced_question.data_source
+    ):
+        referenced_data_source_items = get_referenced_data_source_items_by_managed_expression(
+            managed_expression=managed_expression
+        )
+        for dsir in expression.data_source_item_references:
+            db.session.delete(dsir)
+        expression.data_source_item_references = [
+            DataSourceItemReference(expression_id=expression.id, data_source_item_id=referenced_data_source_item.id)
+            for referenced_data_source_item in referenced_data_source_items
+        ]
```

Comment:
nit: Since this is copy/pasted below, possibly this could all be a little `_update_data_source_references` helper?